### PR TITLE
Sphinx documentation

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Helps resolve [SIMPLY-2161](https://jira.nypl.org/browse/SIMPLY-2161).

The [documentation on readthedocs](https://circulation.readthedocs.io/en/sphinx-documentation/index.html) could use an introduction and better navigation, but for this first step the goal was to generate the docs and deploy it.

I'll explain more in the PR for `circulation`. For this repo, it's mostly updating the docstrings (in [rst - reStructuredText](http://docutils.sourceforge.net/rst.html)) so that [sphinx](https://www.sphinx-doc.org/en/master/) can parse them correctly. @leonardr also helped fixed a pickle issue in /utils/opds_writer with the lxml package.

### CODE REVIEW

File: coverage.py [hunk 1]
```diff
@@ -516,8 +517,8 @@ def items_that_need_coverage(self, identifiers=None, **kwargs):
         need coverage.
 
         :param subset: A list of Identifier objects. If present, return
-        only items that need coverage *and* are associated with one
-        of these identifiers.
+            only items that need coverage *and* are associated with one
```
Comment (EdwinGuzman) [inline coverage.py]: Any text related to a param should be indented.

File: external_search.py [hunk 2]
```diff
@@ -819,8 +819,8 @@ def version_name(cls):
     def script_name(cls, base_name):
         """Scope a script name with "simplified" (to avoid confusion with
         other applications on the Elasticsearch server), and the
-        version number (to avoid confusion with other versions _of
-        this application_, which may implement the same script
+        version number (to avoid confusion with other versions *of
```
Comment (EdwinGuzman) [inline external_search.py]: Emphasis in rst uses * instead of _.

File: opds.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 import requests
 
-from lxml import builder, etree
```
Comment (EdwinGuzman) [inline opds.py]: Not used so removed, also in other files.

File: testing.py [hunk 4]
```diff
@@ -754,13 +756,16 @@ def print_database_class(cls, db_connection):
...
         vast tracts of data into your output stream on production.
 
-        Call like this:
-        set_trace()
-        from testing import (
-            DatabaseTest,
-        )
-        _db = Session.object_session(self)
-        DatabaseTest.print_database_class(_db)  # TODO: remove before prod
+        Call like this::
```
Comment (EdwinGuzman) [inline testing.py]: Code snippets need to go after `::` and a new paragraph. End result:
Screen Shot 2019-07-31 at 3 28 40 PM

File: util/opds_writer.py [hunk 5]
```diff
@@ -5,6 +5,16 @@
 from lxml import builder, etree
 from nose.tools import set_trace
 
+class ElementMaker(builder.ElementMaker):
```
Comment (EdwinGuzman) [inline util/opds_writer.py]: @leonardr helped fixed this pickle issue:
Screen Shot 2019-07-30 at 5 02 23 PM

**Review by aslagle — APPROVED** [review]

**Review by adriana-alter — APPROVED** [review]

File: model/configuration.py [hunk 6]
```diff
@@ -266,13 +266,14 @@ def with_setting_value(cls, _db, protocol, goal, key, value):
...
         ExternalIntecation itself.
+
         :param protocol: ExternalIntegrations must have this protocol.
         :param goal: ExternalIntegrations must have this goal.
         :param key: Look only at ExternalIntegrations with
             a ConfigurationSetting for this key.
         :param value: Find ExternalIntegrations whose ConfigurationSetting
             has this value.
-        :return: A Query object.
+        :returns: A Query object.
```
Comment (leonardr) [inline model/configuration.py]: I think `:return:` is correct -- it's all over this code.

**Review by leonardr — APPROVED** [review]

File: model/configuration.py (hunk 6 above)
Comment (EdwinGuzman) [reply model/configuration.py]: Either one works, but I'll change to `:return:` for consistency.

**Pull request merged**